apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -318,7 +318,7 @@ get_one_of_path(_Elem, [], Def) ->
 is_archivable_message(Mod, Dir, Packet=#xmlel{name = <<"message">>}) ->
     Type = xml:get_tag_attr_s(<<"type">>, Packet),
     is_valid_message_type(Mod, Dir, Type) andalso
-        is_valid_message(Mod, Dir, Packet);
+        (is_valid_message(Mod, Dir, Packet) orelse is_chat_marker(Packet));
```

Comment:
I've just noticed that this change will cause ALL messages with markers to be archived. Including the ones with "delay" or "no-store" elements, so e.g. Stream Management retransmissions would get stored, even though they must not be.